packages/openapi3/test/emit-openapi.test.ts

```diff
@@ -1,79 +1,136 @@
-        True: "True",
-        TRUE: "TRUE",
-        tRUE: "tRUE",
-        "false": "false",
-        False: "False",
-        FALSE: "FALSE",
-        fALSE: "fALSE",
-        on: "on",
-        On: "On",
-        ON: "ON",
-        oN: "oN",
-        off: "off",
-        Off: "Off",
-        OFF: "OFF",
-        oFF: "oFF"
-      }
-      `);
-    expect(content).toBe(`openapi: 3.0.0
-info:
-  title: (title)
-  version: 0.0.0
-tags: []
-paths: {}
-components:
-  schemas:
-    TestEnum:
-      type: string
-      enum:
-        - 'y'
-        - 'Y'
-        - 'yes'
-        - 'Yes'
-        - 'YES'
-        - yEs
-        - 'n'
-        - 'N'
-        - 'no'
-        - 'No'
-        - 'NO'
-        - nO
-        - 'true'
-        - 'True'
-        - 'TRUE'
-        - tRUE
-        - 'false'
-        - 'False'
-        - 'FALSE'
-        - fALSE
-        - 'on'
-        - 'On'
-        - 'ON'
-        - oN
-        - 'off'
-        - 'Off'
-        - 'OFF'
-        - oFF
-`);
+  worksFor(["3.0.0", "3.1.0"], (helper) => {
+    it("should add single quote for y|Y|yes|Yes|YES|n|N|no|No|NO|true|True|TRUE|false|False|FALSE|on|On|ON|off|Off|OFF", async () => {
+      const [_, __, content] = await helper.emitOpenApiWithDiagnostics(`
+          enum TestEnum {
+            y: "y",
+            Y: "Y",
+            yes: "yes",
+            Yes: "Yes",
+            YES: "YES",
+            yEs: "yEs",
+            n: "n",
+            N: "N",
+            no: "no",
+            No: "No",
+            NO: "NO",
+            nO: "nO",
+            "true": "true",
+            True: "True",
+            TRUE: "TRUE",
+            tRUE: "tRUE",
+            "false": "false",
+            False: "False",
+            FALSE: "FALSE",
+            fALSE: "fALSE",
+            on: "on",
+            On: "On",
+            ON: "ON",
+            oN: "oN",
+            off: "off",
+            Off: "Off",
+            OFF: "OFF",
+            oFF: "oFF"
+          }
+          `);
+      expect(content).toBe(`openapi: 3.0.0
+    info:
+      title: (title)
+      version: 0.0.0
+    tags: []
+    paths: {}
+    components:
+      schemas:
+        TestEnum:
+          type: string
+          enum:
+            - 'y'
+            - 'Y'
+            - 'yes'
+            - 'Yes'
+            - 'YES'
+            - yEs
+            - 'n'
+            - 'N'
+            - 'no'
+            - 'No'
+            - 'NO'
+            - nO
+            - 'true'
+            - 'True'
+            - 'TRUE'
+            - tRUE
+            - 'false'
+            - 'False'
+            - 'FALSE'
+            - fALSE
+            - 'on'
+            - 'On'
+            - 'ON'
+            - oN
+            - 'off'
+            - 'Off'
+            - 'OFF'
+            - oFF
+    `);
+    });
+
+    interface ServiceNameCase {
+      description: string;
+      code: string;
+      outputFilePattern: string;
+      expectedOutputFiles: string[];
+    }
+    it.each([
```

Comment:
those test don't seem like they belong to this file and definitely not to this describe